sklearn/feature_selection/tests/test_rfe.py

```diff
@@ -64,8 +65,9 @@ def test_rfe_set_params():
     y_pred = rfe.fit(X, y).predict(X)
 
     clf = SVC()
-    rfe = RFE(estimator=clf, n_features_to_select=4, step=0.1,
-              estimator_params={'kernel': 'linear'})
+    with warnings.catch_warnings(record=True):
```

Comment:
I think we should explicit in a comment the kind of warnings we expect to catch in such tests and why we catch them instead of addressing the warnings (sometimes it's not possible).